pkgs/native_toolchain_c/lib/src/cbuilder/linker_options.dart

```diff
@@ -152,21 +170,33 @@ extension LinkerOptionsExt on LinkerOptions {
   ) => [
     ...sourceFiles,
     '/link',
-    if (_includeAllSymbols) ...sourceFiles.map((e) => '/WHOLEARCHIVE:$e'),
+    if (_keepAllSymbols) ...sourceFiles.map((e) => '/WHOLEARCHIVE:$e'),
     ..._linkerFlags,
-    ..._symbolsToKeep?.map(
-          (symbol) =>
-              '/INCLUDE:${targetArch == Architecture.ia32 ? '_' : ''}$symbol',
-        ) ??
-        [],
-    if (linkerScript != null)
-      '/DEF:${linkerScript!.toFilePath()}'
-    else if (_generateLinkerScript && _symbolsToKeep != null)
-      '/DEF:${_createClLinkScript(_symbolsToKeep)}',
+    ..._symbols.map(
+      (symbol) =>
+          '/INCLUDE:${targetArch == Architecture.ia32 ? '_' : ''}$symbol',
+    ),
+    if (_linkerScriptMode is ManualLinkerScript)
+      '/DEF:${_linkerScriptMode.script.toFilePath()}'
+    else if (_linkerScriptMode is GenerateLinkerScript)
+      '/DEF:${_createClLinkScript(_symbols)}',
     if (stripDebug) '/PDBSTRIPPED',
     if (gcSections) '/OPT:REF',
   ];
 
+  /// This creates a list of exported symbols.
+  ///
+  /// If this is not set, some symbols might be kept. This can be inspected
+  /// using `ld -why_live`, see https://www.unix.com/man_page/osx/1/ld/, where
+  /// the reason will show up as `global-dont-strip`.
```

Comment:
Leave a breadcrumb here that this seems to be a rust thing?

Reply:
(If it is, we still don't really know...)